relay/cmd/lib/relay.go

```diff
@@ -3,84 +3,69 @@ package lib
 import (
 	"context"
 	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
 
-	"github.com/Layr-Labs/eigenda/common"
-	"github.com/Layr-Labs/eigenda/common/aws/dynamodb"
-	"github.com/Layr-Labs/eigenda/common/aws/s3"
-	"github.com/Layr-Labs/eigenda/common/geth"
-	"github.com/Layr-Labs/eigenda/core/eth"
-	"github.com/Layr-Labs/eigenda/core/thegraph"
-	"github.com/Layr-Labs/eigenda/disperser/common/v2/blobstore"
 	"github.com/Layr-Labs/eigenda/relay"
-	"github.com/Layr-Labs/eigenda/relay/chunkstore"
-	gethcommon "github.com/ethereum/go-ethereum/common"
-	"github.com/prometheus/client_golang/prometheus"
 	"github.com/urfave/cli"
 )
 
 // RunRelay is the entrypoint for the relay.
-func RunRelay(ctx *cli.Context) error {
-	config, err := NewConfig(ctx)
+func RunRelay(cliCtx *cli.Context) error {
+	config, err := NewConfig(cliCtx)
 	if err != nil {
 		return fmt.Errorf("failed to create relay config: %w", err)
 	}
 
-	logger, err := common.NewLogger(&config.Log)
-	if err != nil {
-		return fmt.Errorf("failed to create logger: %w", err)
-	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	dynamoClient, err := dynamodb.NewClient(config.AWS, logger)
+	// Create relay server dependencies
+	deps, err := relay.NewServerDependencies(
+		ctx,
+		relay.ServerDependenciesConfig{
+			AWSConfig:                  config.AWS,
+			MetadataTableName:          config.MetadataTableName,
+			BucketName:                 config.BucketName,
+			OperatorStateRetrieverAddr: config.OperatorStateRetrieverAddr,
+			ServiceManagerAddr:         config.EigenDAServiceManagerAddr,
+			ChainStateConfig:           config.ChainStateConfig,
+			EthClientConfig:            config.EthClientConfig,
+			LoggerConfig:               config.Log,
+		},
```

Comment:
why isnt this already a substruct of the config?

Reply:
reverted after discussion https://github.com/Layr-Labs/eigenda/pull/2107/commits/a8c93ca0c24cfb0d2bea58b847e45a20ab69c91c